server/src/main/java/org/elasticsearch/rollup/RollupShardDecider.java

```diff
@@ -0,0 +1,239 @@
+            return true;
+        }
+        // A rollup index is being searched
+        if (checkRollupConditions(request) == false) {
+            return false;
+        }
+
+        final AggregatorFactories.Builder aggregations = request.source() != null ? request.source().aggregations() : null;
+        final String rollupConfigSource = indexRollupMetadata.get(RollupIndexMetadata.ROLLUP_META_FIELD);
+        RollupIndexMetadata rollupIndexMetadata = RollupIndexMetadata.parseMetadataXContent(rollupConfigSource);
+
+        DateHistogramAggregationBuilder source = getDateHistogramAggregationBuilder(aggregations);
+        ZoneId sourceTimeZone = ZoneOffset.UTC; // Default timezone is UTC
+        if (source != null && source.timeZone() != null) {
+            sourceTimeZone = ZoneId.of(source.timeZone().toString(), ZoneId.SHORT_IDS);
+        }
+        DateHistogramInterval sourceInterval = source != null ? source.getCalendarInterval() : null;
+        // Incompatible timezone => skip this rollup group
+        if (canMatchTimezone(sourceTimeZone, rollupIndexMetadata.getDateTimezone().zoneId()) == false
+            || canMatchCalendarInterval(sourceInterval, rollupIndexMetadata.getDateInterval()) == false) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Parses the search request and checks if the required conditions are met so that results
+     * can be answered by a rollup index.
+     *
+     * Conditions that must be met are:
+     *  - Request size must be 0
+     *  - Metric aggregations must be supported
+     *  - No use of runtime fields in the query
+     *
+     * @param request the search request to parse
+     * @return true if a rollup index can
+     */
+    private static boolean checkRollupConditions(ShardSearchRequest request) {
+        if (request.source() == null) {
+            return false;
+        }
+
+        // If request size is not 0, rollup indices should not match
+        if (request.source().size() > 0) {
+            return false;
+        }
+
+        // Check for supported aggs
+        AggregatorFactories.Builder aggregations = request.source().aggregations();
+        if (checkSupportedAggregations(aggregations) == false){
+            return false;
+        }
+
+        // TODO(csoulios): Add check for runtime fields in the request
+        return true;
+    }
+
+    /**
+     * Check if requested aggregations are supported by rollup indices
+     *
+     * @param aggregations the aggregation builders
+     * @return true if aggregations are supported by rollups, otherwise false
+     */
+    private static boolean checkSupportedAggregations(AggregatorFactories.Builder aggregations) {
+        if (aggregations == null) {
+            return false;
+        }
+
+        for (AggregationBuilder builder : aggregations.getAggregatorFactories()) {
+            if (SUPPORTED_AGGS.contains(builder.getWriteableName()) == false) {
```

Comment:
We have to check date histogram that appears as children. It would be nice if the logic could be done in the aggregation directly though. It should be the responsibility of the aggregation to validate the field usage rather than this high level check. That's fragile and error-prone, we need something that is sustainable for the long term.